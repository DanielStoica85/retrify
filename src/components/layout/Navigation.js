import React, { Component } from 'react';
import {
    Collapse,
    Navbar,
    NavbarToggler,
    NavbarBrand,
    Nav,
    Container
} from 'reactstrap';

import SignedInLinks from './SignedInLinks';
import SignedOutLinks from './SignedOutLinks';

import { connect } from 'react-redux';

export class Navigation extends Component {
    state = {
        isOpen: false
    };

    toggle = () => {
        this.setState({
            isOpen: !this.state.isOpen
        });
    };

    render() {
        const { auth } = this.props;
        const links = auth.uid ? <SignedInLinks /> : <SignedOutLinks />;
        return (
            <div>
                <Navbar color="dark" dark expand="sm" className="mb-5">
                    <Container>
                        <NavbarBrand href="/">Retrify</NavbarBrand>
                        <NavbarToggler onClick={this.toggle} />
                        <Collapse isOpen={this.state.isOpen} navbar>
                            <Nav className="ml-auto" navbar>
                                {links}
                            </Nav>
                        </Collapse>
                    </Container>
                </Navbar>
            </div>
        );
    }
}

const mapStateToProps = state => {
    return {
        auth: state.firebase.auth
    };
};

export default connect(mapStateToProps)(Navigation);
