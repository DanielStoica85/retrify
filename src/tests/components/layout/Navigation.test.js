import React from 'react';
import { shallow } from 'enzyme';
import Navigation from '../../../components/layout/Navigation';
import { NavbarToggler } from 'reactstrap';

let component;

describe('Navigation component', () => {
    beforeEach(() => {
        component = shallow(<Navigation />);
    });

    it('should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('should set isOpen on  navbarToggler click', () => {
        component.find(NavbarToggler).simulate('click');
        expect(component.state('isOpen')).toEqual(true);
        expect(component).toMatchSnapshot();
    });
});
