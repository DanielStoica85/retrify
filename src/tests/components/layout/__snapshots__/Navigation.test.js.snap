// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should render correctly 1`] = `
<div>
  <Navbar
    className="mb-5"
    color="dark"
    dark={true}
    expand="sm"
    tag="nav"
  >
    <Container
      tag="div"
    >
      <NavbarBrand
        href="/"
        tag="a"
      >
        Retrify
      </NavbarBrand>
      <NavbarToggler
        onClick={[Function]}
        tag="button"
        type="button"
      />
      <Collapse
        appear={false}
        enter={true}
        exit={true}
        in={false}
        isOpen={false}
        mountOnEnter={false}
        navbar={true}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        tag="div"
        timeout={350}
        unmountOnExit={false}
      >
        <Nav
          className="ml-auto"
          navbar={true}
          tag="ul"
          vertical={false}
        >
          <SignedOutLinks />
          <SignedInLinks />
        </Nav>
      </Collapse>
    </Container>
  </Navbar>
</div>
`;

exports[`Navigation component should set isOpen on  navbarToggler click 1`] = `
<div>
  <Navbar
    className="mb-5"
    color="dark"
    dark={true}
    expand="sm"
    tag="nav"
  >
    <Container
      tag="div"
    >
      <NavbarBrand
        href="/"
        tag="a"
      >
        Retrify
      </NavbarBrand>
      <NavbarToggler
        onClick={[Function]}
        tag="button"
        type="button"
      />
      <Collapse
        appear={false}
        enter={true}
        exit={true}
        in={false}
        isOpen={true}
        mountOnEnter={false}
        navbar={true}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        tag="div"
        timeout={350}
        unmountOnExit={false}
      >
        <Nav
          className="ml-auto"
          navbar={true}
          tag="ul"
          vertical={false}
        >
          <SignedOutLinks />
          <SignedInLinks />
        </Nav>
      </Collapse>
    </Container>
  </Navbar>
</div>
`;
